core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -428,4 +459,25 @@ private void notifyCreateEvent(boolean isDirectory, Path path, int count, Path r
     }
     onEvent(EventType.CREATE, isDirectory, path, count, rootPath);
   }
+
+  private class OnIdleTimerTask extends TimerTask {
+
+    private boolean isDone = false;
+
+    private int counter;
+
+    private DirectoryChangeListener listener;
+
+    private OnIdleTimerTask(DirectoryChangeListener listener, int counter) {
+      this.listener = listener;
+      this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+      logger.debug("Task performed on: " + new Date() + " with event count " + counter);
+      listener.onIdle(counter);
```

Comment:
We should document that this operation is not atomic - i.e. the check on isDone() and cancel() is not atomic and may overlap with this sections onIdle() call and setting isDone. And thus it is possible that actually the timer is cancelled, while this event is being fired. I would imagine for the type of use cases this is being applied to, that is not a problem, but either way should be in in the JavaDocs.